utils/utils.go

```diff
@@ -160,3 +160,16 @@ func PopulateMetadataForBulkPublishEntry(reqMeta, entryMeta map[string]string) m
 
 	return resMeta
 }
+
+func Filter[T any](items []T, fn func(item T) bool, isFilterNotApplicable bool) []T {
+	if isFilterNotApplicable {
+		return items
+	}
+	filteredItems := []T{}
+	for _, value := range items {
+		if fn(value) {
+			filteredItems = append(filteredItems, value)
+		}
+	}
+	return filteredItems
```

Comment:
```suggestion
	filteredItems := make([]T, 0, len(items))
	for _, value := range items {
		if fn(value) {
			filteredItems = append(filteredItems, value)
		}
	}
	return filteredItems[:len(filteredItems):len(filteredItems)] // return and trunc capacity of slice
```

Reply:
`return filteredItems[:len(filteredItems)] // return the truncated slice`

Reply:
@ItalyPaleAle when i tried with this, I was getting some failures here ... when running multiple unit tests in series, even though the type is only slice and not pointer to slice, changing the values in place was affecting the source !

Reply:
Used array with a capacity pre-defined. But, as discussed, not doing in-place.